### ISSUE TITLE
Bug: env variable database.default.dsn not working for mysqli

### ISSUE THREAD

**Issue by ankahla** [codeigniter4/CodeIgniter4#3456]
**Describe the bug**
When we set a correct mysql DSN in the file .env we got an exception "_Unable to connect to the database_".
Example:
`database.default.DSN=mysql://b4ee2ad4fc835b:b9af4360@us-cdbr-east-02.cleardb.com/heroku_9672675f9566574`

**CodeIgniter 4 version**
4.0.4

**Affected module(s)**
https://github.com/codeigniter4/CodeIgniter4/blob/develop/system/Database/MySQLi/Connection.php#L93

**Expected behavior, and steps to reproduce if appropriate**
When we set a value for the variable "database.default.dsn" in the.env file we should not get the error "_Unable to connect to the database_" and we should be able to connect to the database.

**Context**
 - OS: php:7.4-fpm
 - Web server : Docker image httpd:2.4
 - PHP : 7.4

**Comment by michalsn** [comment]
Yes, I'm afraid DSN string support is not implemented in MySQLi and SQLite3. PR would be welcomed!
